### PR TITLE
Issue #47 Added places API

### LINKED ISSUE #47: [API Request] Places API
Call [Here Places API](https://developer.here.com/documentation/places/topics/quick-start-find-text-string.html) to fetch results.
Cache results to serve them faster ([Reference Code](https://github.com/project-travel-mate/server/blob/master/api/modules/city/views.py#L15))
### Input Parameters
+ latitude
+ longitude
+ category_tags

### Output (a list of object, each containing following) 
+ title
+ vicinity (address)
+ href (website)
+ icon

### PR DESCRIPTION
**Pull request by techac**
# Description

added places api issue #47 

Fixes #47 

## Type of change
Just put an x in the [] which are valid.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.
- [x] `flake8`
- [x] `python manage.py test`

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

**Comment by techac** [comment]
@prabhakar267 PR is ready for review

File: api/urls.py [hunk 1]
```diff
@@ -13,6 +13,7 @@
 from api.modules.github import views as github_views
 from api.modules.twitter import views as twitter_views
 from api.modules.food import views as food_views
+from api.modules.PlacesAPI import views as places_views
```
Comment (prabhakar267) [inline api/urls.py]: Change module name to `hyperlocal`

File: api/modules/PlacesAPI/views.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        response = []
+        suggestions = api_response_json['results']
+        for place in suggestions:
+            result = PlaceResponse(
+                title=place['title'],
+                href=place['href']
+            )
+            result_as_json = result.to_json()
+            response.append(result_as_json)
+    except Exception as e:
```
Comment (prabhakar267) [inline api/modules/PlacesAPI/views.py]: Have different exceptions for 
+ incorrect / expired API keys
+ throttling error (if possible)

File: api/modules/PlacesAPI/constants.py [hunk 3]
```diff
@@ -0,0 +1,11 @@
+"""
+    Use Places API to access locations using longitude and latitude
+"""
+import os
+
+BASE_URL = 'https://places.cit.api.here.com/places/v1/autosuggest?'
+APP_ID = os.environ.get("YOUR_APP_ID", "")
+APP_CODE = os.environ.get("YOUR_APP_CODE", "")
+
+PLACES_API_SEARCH = BASE_URL + 'at={latitude},{longitude}&q={place_keyword}' + '&app_id=' + APP_ID + '&app_code=' + \
```
Comment (prabhakar267) [inline api/modules/PlacesAPI/constants.py]: `PLACES_SEARCH_API_URL`

File: api/modules/PlacesAPI/constants.py [hunk 4]
```diff
@@ -0,0 +1,11 @@
+"""
+    Use Places API to access locations using longitude and latitude
+"""
+import os
+
+BASE_URL = 'https://places.cit.api.here.com/places/v1/autosuggest?'
```
Comment (prabhakar267) [inline api/modules/PlacesAPI/constants.py]: `PLACES_API_BASE_URL`

File: api/modules/PlacesAPI/constants.py [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+APP_ID = os.environ.get("YOUR_APP_ID", "")
```
Comment (prabhakar267) [inline api/modules/PlacesAPI/constants.py]: use better names than `APP_ID`, `YOUR_APP_ID`, `APP_CODE` and `YOUR_APP_CODE`

File: api/modules/PlacesAPI/response.py [hunk 6]
```diff
@@ -0,0 +1,15 @@
+class PlaceResponse(object):
```
Comment (prabhakar267) [inline api/modules/PlacesAPI/response.py]: `HyperlocalResponse`
and filename should be `hyperlocal_response.py`

File: api/modules/PlacesAPI/response.py [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+    """
+    Specifies the response to be sent foo one place object.
+    """
+    def __init__(self, *args, **kwargs):
+        self.title = kwargs.get('title')
+        self.vicinty = kwargs.get('vicinity')
```
Comment (prabhakar267) [inline api/modules/PlacesAPI/response.py]: Change name to `address`

File: api/modules/PlacesAPI/response.py [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+        self.href = kwargs.get('href')
+
```
Comment (prabhakar267) [inline api/modules/PlacesAPI/response.py]: Add icon as well

File: api/modules/PlacesAPI/views.py [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+        if api_response.status_code == 503:
+            error_message = "Service Unavailable"
+            return Response(error_message, status=status.HTTP_503_SERVICE_UNAVAILABLE)
+        response = []
+        suggestions = api_response_json['results']
+        for place in suggestions:
+            result = PlaceResponse(
+                title=place['title'],
+                href=place['href']
+            )
```
Comment (prabhakar267) [inline api/modules/PlacesAPI/views.py]: Add icon and address

**Review by prabhakar267 — CHANGES_REQUESTED** [review]

File: api/modules/PlacesAPI/views.py [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+        suggestions = api_response_json['results']
+        for place in suggestions:
+            result = HyperLocalResponse(
+                title=place['title'],
+                website=place['href'],
+                address=place['vicinity'],
+            )
+            result_as_json = result.to_json()
+            response.append(result_as_json)
+    except Exception as e:
```
Comment (prabhakar267) [inline api/modules/PlacesAPI/views.py]: If `place_keyword` keyword is incorrect, send an appropriate response

Comment (prabhakar267) [inline api/modules/PlacesAPI/views.py]: If API keys are incorrect or missing, send an appropriate response

**Review by prabhakar267 — CHANGES_REQUESTED** [review]

**Comment by techac** [comment]
@prabhakar267 you may review PR

**Review by prabhakar267 — APPROVED** [review]

**Pull request merged**